### DISCUSSION TITLE
强化后退役能否改成 退役后再强化

### DISCUSSION THREAD

**Discussion by ddr850814** [LmeSzinc/AzurLaneAutoScript#1096] [Q&A] [answered]
一键退役很快 但是强化很慢 要等强化到没材料了才一键退役 那很费事
而且白船是出的最多的要想白船满了去强化很容易 要想蓝船满了去退役就很难 最后没剩几个格子的时候不停尝试强化
并且不会跳过在编队里的船

**Comment by Horizon101011** [comment]
> 并且不会跳过在编队里的船

新的强化抛弃了原有的使用消息弹窗来区分状态的方式，使用了更稳定的特征进行区分
因此，状态的区分最早会出现在点击推荐的时候
如果点击推荐之后没有检测到强化素材，将会直接判定为强化失败然后尝试下一个
只有出现素材才会尝试强化从而区分成功或者战斗中
所以强化失败的优先级是高于战斗中的
在没有可用素材的情况下，跳过编队中的船只会单纯增加尝试的次数，所以这里不作跳过处理

当然这里也会有一点缺陷，如果只是单纯因为编队中的船需要的素材缺失，确实会存在仍有强化的可能但是直接退役的情况
所以这种情况建议稍微增大一点点强化数量的设置
>最后没剩几个格子的时候不停尝试强化

至于这一点，确实是一个待优化的问题

**Comment by LmeSzinc** [comment, marked as answer]
已在 #1117 中修改
